### PR TITLE
Add act support

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

Add Act support.

##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

**Comment by Hoverbear** [comment]
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,8 @@ jobs:
           logger: pretty
           log-directives: nix_installer=trace
           backtrace: full
+          nix-installer-branch: main
+      - run: cat /etc/nix/nix.conf
```
Comment (cole-h) [inline .github/workflows/ci.yml:21]:
```suggestion
```

I think

**Review by cole-h — APPROVED** [review]
Else LGTM, but I don't use ACT so I can't test it.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Hoverbear) [reply .github/workflows/ci.yml:21]: Ah ya...

**Comment by Hoverbear** [comment]
@cole-h I was able to run `nix run nixpkgs#act` in the directory if I had `services.docker.enable` set.

**Review by lheckemann — APPROVED** [review]

**Comment by lheckemann** [comment]
Changes LGTM but it doesn't actuallly work for me:
```

[linus@oak:~/scratch/nix-installer-action]$ nix run nixpkgs#act
[CI/Run x86_64 Linux ] 🚀  Start image=node:16-bullseye-slim
[CI/Run x86_64 Darwin] 🚧  Skipping unsupported platform -- Try running with `-P macos-12=...`
[CI/Run x86_64 Linux ]   🐳  docker pull image=node:16-bullseye-slim platform= username= forcePull=true
[CI/Run x86_64 Linux ]   🐳  docker create image=node:16-bullseye-slim platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[CI/Run x86_64 Linux ]   🐳  docker run image=node:16-bullseye-slim platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[CI/Run x86_64 Linux ] ⭐ Run Main actions/checkout@v3
[CI/Run x86_64 Linux ]   🐳  docker cp src=/home/linus/scratch/nix-installer-action/. dst=/home/linus/scratch/nix-installer-action
[CI/Run x86_64 Linux ]   ✅  Success - Main actions/checkout@v3
[CI/Run x86_64 Linux ] ⭐ Run Main Install Nix
[CI/Run x86_64 Linux ] ⭐ Run Main Install Nix
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/1-composite-0.sh] user= workdir=
| Detected `$ACT` environment, assuming this is a https://github.com/nektos/act created container, set `NOT_ACT=true` to override this. This
 will change the settings of the `init` as well as `extra-conf` to be compatible with `act`
| Set NIX_INSTALLER_EXTRA_CONF=sandbox = false
| trusted-users = root
| Set NIX_INSTALLER_URL=https://install.determinate.systems/nix/branch/main
| Set NIX_INSTALLER_LOGGER=pretty
| Set NIX_INSTALLER_LOG_DIRECTIVES=nix_installer=trace
| Set RUST_BACKTRACE=full
| Set NIX_INSTALLER_DIAGNOSTIC_ENDPOINT=https://install.determinate.systems/nix/diagnostic
| Set NIX_INSTALLER_NO_CONFIRM=true
| /var/run/act/workflow/1-composite-0.sh: line 208: curl: command not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Install Nix
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ]   ❌  Failure - Main Install Nix
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/3] user= workdir=
| /var/run/act/workflow/3: line 2: nix: command not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Main Test bash
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[bash --login /var/run/act/workflow/4] user= workdir=
| /var/run/act/workflow/4: line 2: nix-instantiate: command not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Test bash
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Main Test sh
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[sh -l /var/run/act/workflow/5] user= workdir=
| /var/run/act/workflow/5: 2: nix-instantiate: not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Test sh
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/7] user= workdir=
| /var/run/act/workflow/7: line 2: nix: command not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/9] user= workdir=
| /var/run/act/workflow/9: line 2: nix: command not found
[CI/Run x86_64 Linux ]   ❌  Failure - Main Test `nix` with `$GITHUB_PATH`
[CI/Run x86_64 Linux ] exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
[CI/Run x86_64 Linux ] ⭐ Run Post Install Nix
[CI/Run x86_64 Linux ]   ✅  Success - Post Install Nix
[CI/Run x86_64 Linux ] 🏁  Job failed
Error: Job 'Run x86_64 Linux' failed
```
am I doing a stupid here?

**Comment by Hoverbear** [comment]
I don't think you are!

**Comment by Hoverbear** [comment]
@lheckemann I think I found the problem and it's working locally for me now. Can you try again?

**Comment by lheckemann** [comment]
Still getting the same thing :/

**Comment by Hoverbear** [comment]
I don't understand how your image has no curl and mine does...

**Comment by Hoverbear** [comment]
Why does yours pull `docker pull image=node:16-bullseye-slim`?

**Comment by lheckemann** [comment]
I have no idea :D what does yours use?

**Comment by Hoverbear** [comment]
```
❯ sudo nix run nixpkgs#act
[CI/Run x86_64 Darwin] 🚧  Skipping unsupported platform -- Try running with `-P macos-12=...`
[CI/Run x86_64 Linux ] 🚀  Start image=catthehacker/ubuntu:act-22.04
[CI/Run x86_64 Linux ]   🐳  docker pull image=catthehacker/ubuntu:act-22.04 platform= username= forcePull=true
[CI/Run x86_64 Linux ]   🐳  docker create image=catthehacker/ubuntu:act-22.04 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
[CI/Run x86_64 Linux ]   🐳  docker run image=catthehacker/ubuntu:act-22.04 platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
```

I assume our `.actrc` are different: https://github.com/nektos/act#first-act-run

```
❯ cat ~/.actrc 
-P ubuntu-latest=catthehacker/ubuntu:act-latest
-P ubuntu-22.04=catthehacker/ubuntu:act-22.04
-P ubuntu-20.04=catthehacker/ubuntu:act-20.04
-P ubuntu-18.04=catthehacker/ubuntu:act-18.04
```

**Comment by lheckemann** [comment]
Aaah yes, using the nixos-unstable branch of nixpkgs (was using 23.05 before) after removing actrc does the trick!

**Pull request merged**